app/helpers/api_helper.rb

```diff
@@ -8,127 +8,94 @@
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:to_i)) &&
+      xml.tag!(tag, str.to_s, type: "integer")
   end
 
   def xml_float(xml, tag, val, places)
-    str = begin
-            "%.#{places}f" % val
-          rescue StandardError
-            ""
-          end
-    xml.tag!(tag, str, type: "float")
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:round, places)) &&
+      xml.tag!(tag, str.to_s, type: "float")
   end
 
   def xml_string(xml, tag, val)
-    if val.present?
-      str = val.to_s
-      xml.tag!(tag, str, type: "string", content_type: "text/plain")
-    end
-  rescue StandardError
+    val.present? &&
+      xml.tag!(tag, val.to_s, type: "string", content_type: "text/plain")
   end
 
   def xml_html_string(xml, tag, val)
-    if val.present?
-      str = val.to_s
-      xml.tag!(tag, str, type: "string", content_type: "text/html")
-    end
-  rescue StandardError
+    val.present? &&
+      xml.tag!(tag, val.to_s, type: "string", content_type: "text/html")
   end
 
   def xml_sql_string(xml, tag, val)
-    if val.present?
-      str = val.to_s
-      xml.tag!(tag, str, type: "string", content_type: "application/x-sql")
-    end
-  rescue StandardError
+    val.present? &&
+      xml.tag!(tag, val.to_s, type: "string", content_type: "application/x-sql")
   end
 
   def xml_date(xml, tag, val)
-    if val
-      str = begin
-              val.api_date
-            rescue StandardError
-              ""
-            end
+    val.present? &&
+      (str = val.try(:api_date)) &&
       xml.tag!(tag, str, type: "date", format: "YYYY-MM-DD")
-    end
-  rescue StandardError
   end
 
   def xml_datetime(xml, tag, val)
-    if val
-      str = val.utc.api_time
+    val.present? &&
+      (str = val.try(:utc).try(:api_time)) &&
       xml.tag!(tag, str, type: "date-time", format: "YYYY-MM-DD HH:MM:SS")
-    end
-  rescue StandardError
   end
 
   def xml_ellapsed_time(xml, tag, val)
-    str = "%.4f" % val
-    xml.tag!(tag, str, type: "float", units: "seconds")
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:round, 4)) &&
+      xml.tag!(tag, str.to_s, type: "float", units: "seconds")
   end
 
   def xml_latitude(xml, tag, val)
-    str = "%.4f" % val
-    xml.tag!(tag, str, type: "float", units: "degrees north")
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:round, 4)) &&
+      xml.tag!(tag, str.to_s, type: "float", units: "degrees north")
   end
 
   def xml_longitude(xml, tag, val)
-    str = "%.4f" % val
-    xml.tag!(tag, str, type: "float", units: "degrees east")
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:round, 4)) &&
+      xml.tag!(tag, str.to_s, type: "float", units: "degrees east")
   end
 
   def xml_altitude(xml, tag, val)
-    str = "%d" % val
-    xml.tag!(tag, str, type: "integer", units: "meters")
-  rescue StandardError
-  end
-
-  def xml_undefined_location(xml, tag, val)
```

Comment:
Don't understand why this is going away.  Again it seems like a potential change to the output of the API call.  I *think* this might be redundant given that the `:location` is specified at least for observations, but I'm not sure.  Again I don't know how the API is really being used, but I would caution against changing the output.

Reply:
It was superceded by xml_minimal_location and xml_detailed_location.  I forget what the issue was that required the change.